From 5136c399de881ef04d141cccb12d605931d0a3dc Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:06:14 -0400
Subject: [usb] uas: Fix uas not working when plugged into an ehci port

Message-id: <1410887463-20674-303-git-send-email-dzickus@redhat.com>
Patchwork-id: 95320
O-Subject: [RHEL7 PATCH 302/591] uas: Fix uas not working when plugged into an ehci port
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 6c2334e9019039d7952190e239e6a8f0d10101fe
Author: Hans de Goede <hdegoede@redhat.com>
Date:   Thu Oct 17 22:20:54 2013 +0200

    uas: Fix uas not working when plugged into an ehci port

    I thought it would be a good idea to also test uas with usb-2, and it turns out
    it was, as it did not work. The problem is that the uas driver was passing the
    bEndpointAddress' direction bit to usb_rcvbulkpipe, the xhci code seems to not
    care about this, but with the ehci code this causes usb_submit_urb failure.

    With this fixed the uas code works nicely with an uas device plugged into
    an ehci port.

    Signed-off-by: Hans de Goede <hdegoede@redhat.com>
    Signed-off-by: Sarah Sharp <sarah.a.sharp@linux.intel.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/storage/uas.c b/drivers/usb/storage/uas.c
index 5eacb80..6ad5de9 100644
--- a/drivers/usb/storage/uas.c
+++ b/drivers/usb/storage/uas.c
@@ -948,13 +948,13 @@ static void uas_configure_endpoints(struct uas_dev_info *devinfo)
   eps[3] = usb_pipe_endpoint(udev, devinfo->data_out_pipe);
  } else {
   devinfo->cmd_pipe = usb_sndbulkpipe(udev,
-      eps[0]->desc.bEndpointAddress);
+          usb_endpoint_num(&eps[0]->desc));
   devinfo->status_pipe = usb_rcvbulkpipe(udev,
-      eps[1]->desc.bEndpointAddress);
+          usb_endpoint_num(&eps[1]->desc));
   devinfo->data_in_pipe = usb_rcvbulkpipe(udev,
-      eps[2]->desc.bEndpointAddress);
+          usb_endpoint_num(&eps[2]->desc));
   devinfo->data_out_pipe = usb_sndbulkpipe(udev,
-      eps[3]->desc.bEndpointAddress);
+          usb_endpoint_num(&eps[3]->desc));
  }
 
  devinfo->qdepth = usb_alloc_streams(devinfo->intf, eps + 1, 3, 256,
-- 
1.7.1